### PR TITLE
ensure ACL'd items are handled correctly by search

### PR DESCRIPTION
**Pull request by thommay**

### CODE REVIEW

File: lib/pedant/rspec/search_util.rb [hunk 1]
```diff
@@ -434,6 +451,20 @@ def can_perform_basic_partial_search_for(object_type, attribute_key, attribute_v
           end
         end # can_perform_basic_partial_search_for
 
+        def can_perform_a_partial_search_that_is_filtered_for(object_type)
```
Comment (marcparadise) [inline lib/pedant/rspec/search_util.rb]: To make the intent clear, let's name with acl: `can_perform_a_partial_search_that_is_acl_filtered_for`

**Comment by marcparadise** [comment]
This looks good, but I'd like to see explicit confirmation that a user who does have the right ACEs can perform the search and see the expected results.  This may just require modifying can_perform_* to accept a requestor argument.

File: spec/api/search/search_spec.rb [hunk 2]
```diff
@@ -71,6 +72,8 @@
 
       test_bad_partial_search_bodies
 
+      can_perform_a_partial_search_that_is_acl_filtered_for :environment
```
Comment (marcparadise) [inline spec/api/search/search_spec.rb]: Should these just be tagged :search?

Comment (thommay) [inline spec/api/search/search_spec.rb]: The `describe` block is tagged, and we don't tag any of the other methods…

Comment (marcparadise) [inline spec/api/search/search_spec.rb]: Yeah, please totally disregard my comment - I scanned and misread badly :D

**Comment by marcparadise** [comment]
:+1:

**Comment by thommay** [comment]
This has been merged

**Pull request closed without merging**